pkg/skaffold/docker/image.go

```diff
@@ -97,6 +106,88 @@ func RunPush(ctx context.Context, cli DockerAPIClient, ref string, out io.Writer
 	return streamDockerMessages(out, rc)
 }
 
+func AddTag(src, target string) error {
+	srcRef, err := resolveReference(src)
+	if err != nil {
+		return errors.Wrap(err, "getting source reference")
+	}
+
+	auth, err := authn.DefaultKeychain.Resolve(srcRef.Context().Registry)
+	if err != nil {
+		return err
+	}
+
+	targetRef, err := resolveReference(target)
+	if err != nil {
+		return errors.Wrap(err, "getting target reference")
+	}
+
+	fmt.Println()
+
+	return addTag(srcRef, targetRef, auth, http.DefaultTransport)
+}
+
+func resolveReference(identifier string) (name.Reference, error) {
+	// try tag first
+	tag, err := name.NewTag(identifier, name.WeakValidation)
+	if err == nil {
+		return tag, nil
+	}
+
+	digest, err := name.NewDigest(identifier, name.WeakValidation)
+	if err == nil {
+		return digest, nil
+	}
+
+	return name.Tag{}, fmt.Errorf("Not a valid tag or digest: %s", identifier)
+}
+
+func addTag(ref name.Reference, targetRef name.Reference, auth authn.Authenticator, t http.RoundTripper) error {
+	tr, err := transport.New(ref, auth, t, transport.PushScope)
+	if err != nil {
+		return err
+	}
+
+	img, err := remote.Image(ref, auth, tr)
+	if err != nil {
+		return err
+	}
+
+	data, err := img.RawManifest()
```

Comment:
Can you use remote.Write here instead of doing the request yourself?
https://github.com/google/go-containerregistry/blob/master/v1/remote/write.go#L40

Reply:
it looks like that calls blobset which would fetch the blobs?

Reply:
It shouldn't. That would be a bug. It looks like BlobSet returns the names of the blobs, which are all checked for existence before upload.